xonsh/fs.py

```diff
@@ -0,0 +1,94 @@
+"""
```

Comment:
Compiling a <100-line file is expensive? The only fresh import I'm seeing is `pathlib` (possibly, depending on how lazy #1862 is), and that becomes a non-issue as more of #2001 is implemented.

The only run-time code executed is trivial (register `Path` as a `PathLike`, and get the system encoding).

After the `pathlib` import, the most expensive thing I'm seeing is the actual parsing and compilation, which isn't something we can lazify. Unless defining an ABC is more involved than I thought?

I can move the other imports into the `except` block, although they should all be already loaded.